hyperloglog_test.go

```diff
@@ -13,29 +13,31 @@ import (
 // Return a dictionary up to n words. If n is zero, return the entire
 // dictionary.
 func dictionary(n int) []string {
-	var words []string
-	dict := "/usr/share/dict/words"
-	f, err := os.Open(dict)
+	path := "/usr/share/dict/words"
+	f, err := os.Open(path)
 	if err != nil {
-		fmt.Printf("can't open dictionary file '%s': %v\n", dict, err)
+		fmt.Printf("can't open dictionary file '%s': %v\n", path, err)
 		os.Exit(1)
 	}
-	count := 0
 	buf := bufio.NewReader(f)
+
+	count := 0
+	words := make([]string, 0, n)
 	for {
-		if n != 0 && count >= n {
+		if n != 0 && count < n {
```

Comment:
If `n <= 0`, won't this loop indefinitely? If `n > 0`, won't this always break on the first iteration?